src/components/ATATPhoneInput.vue

```diff
@@ -0,0 +1,351 @@
+<template>
+  <div id="PhoneControl" class="_atat-phone-field">
+    <div class="d-flex align-center" v-if="label">
+      <label
+        :id="id + '_TextFieldLabel'"
+        class="form-field-label mb-2 mr-2"
+        :for="id + '_TextField'"
+      >
+        {{ label }}
+      </label>
+    </div>
+    <div class="d-flex">
+      <v-select
+        attach
+        id="CountryCodeDropdown'"
+        class="_country-select"
+        :items="searchResults"
+        outlined
+        dense
+        v-model="selectedValue"
+        :height="42"
+        :menu-props="{ bottom: true, offsetY: true }"
+        @change="onChange"
+      >
+        <template v-slot:selection="{ item }">
+          <span class="fi" :class="[`fi-${item.abbreviation}`]"> </span>
+        </template>
+        <template v-slot:prepend-item>
+          <v-text-field
+            v-model="searchTerm"
+            class="_dropdown-text-field"
+            placeholder="Search"
+            @input="searchCountries"
+            append-icon="search"
+            id="DropdownTextField"
+          />
+
+        </template>
+        <template v-slot:item="{ item, on }">
+          <v-list-item class="_country-list"
+                       :class="[item.suggested ? '_suggested' : '', item.active ? '_active' : '']" v-on="on">
+            <v-list-item-content
+              :id="id + '_DropdownListItem_' + item.name.replace(/[^A-Z0-9]/ig, '')"
+              :item-value=item.name
+            >
+              <v-list-item-title class="body _country" >
+                <v-row no-gutters align="center">
+                  <span class=" mr-3 fi" :class="[`fi-${item.abbreviation}`]"> </span>
+                  <span class="mr-2 _country-name" >{{ item.name }}</span>
+                  <span class="color-base body-sm">{{ item.countryCode }}</span>
+                </v-row>
+              </v-list-item-title>
+            </v-list-item-content>
+          </v-list-item>
+        </template>
+      </v-select>
+      <v-text-field
+        ref="atatTextField"
+        :id="id + '_textField'"
+        outlined
+        dense
+        :height="42"
+        :value.sync="_value"
+        :placeholder="placeHolder"
+        @input="phoneMask"
+        class="_phone-number-input"
+        :hide-details="true"
+        :suffix="suffix"
+        :prefix="this.selectedValue.countryCode"
+      >
+      </v-text-field>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from "vue";
+import {Component, Prop, PropSync} from "vue-property-decorator";
+import ATATAutoComplete from "@/components/ATATAutoComplete.vue";
+import ATATTextField from "@/components/ATATTextField.vue";
+import Inputmask from "inputmask/";
```

Comment:
still getting Vetur squiggles on this, and `Instance` lint message about defined but never used
<img width="667" alt="image" src="https://user-images.githubusercontent.com/90333349/160707250-d1c08144-0ed8-43ae-89f4-ea458a96d02c.png">